### DISCUSSION TITLE
Select rows that match part of array element

### DISCUSSION THREAD

**Discussion by Tihomir971** [supabase/supabase#11223] [Questions]
I have products table. Every product can have zero or several different barcodes.
If barcodes are stored in column as text array, how can I filter rows containing part of barcode element?
example: 
name, barcode
"product 1", ["12345","456"]
"product 2", ["52341"]

filter barcode with "234" should return both "product 1" and "product 2"

What is best way to store this kind of values? Tables, json? I like to reduce number of tables if possible.

**Comment by GaryAustin1** [comment]
You don't say how you want do to the query.
If using a client
There is this method https://postgrest.org/en/stable/api.html#computed-virtual-columns to create a computed column.  Then in the function convert your array to a comma separated string in sql.  Then use ilike for your filter on it for a partial string search.

You can also use a generated Postgres column https://www.postgresql.org/docs/current/ddl-generated-columns.html.   You would do the same sql logic to convert the array to a string, and then use ilike filter on this column.

You can also generate the same extra string column using a before trigger on insert/update to generate the extra column.
